src/ring_theory/graded_algebra/basic.lean

```diff
@@ -103,6 +103,42 @@ dfinsupp.mem_support_iff.trans add_submonoid_class.coe_eq_zero.not.symm
 
 end graded_ring
 
+section add_cancel_monoid
+
+open direct_sum dfinsupp finset function
+
+lemma direct_sum.decompose_mul_add_of_left_mem {ι σ A}
+  [decidable_eq ι] [add_left_cancel_monoid ι] [semiring A]
+  [set_like σ A] [add_submonoid_class σ A] (𝒜 : ι → σ) [graded_ring 𝒜]
+  {i j : ι} (a : 𝒜 i) {b : A} :
+  decompose 𝒜 (↑a * b) (i + j) =
+    @graded_monoid.ghas_mul.mul ι (λ i, 𝒜 i) _ _ _ _ a (decompose 𝒜 b j) :=
```

Comment:
Is there anything we can do about this `@graded_monoid.ghas_mul.mul`? It is not very pretty.

cc @ericwieser

Reply:
I think we should add the original version of the lemma too, that uses normal multiplication. It's just the bit of the proof in the argument of `subtype.ext`

Reply:
done